script/IPToken.s.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.18;
+import "forge-std/Script.sol";
+import "../src/IPToken.sol";
+//import "forge-std/console.sol";
+
+//import { ERC1967Proxy } from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+
+contract DeployIPTokenImpl is Script {
+    function run() external {
+        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");
```

Comment:
for "real" scripts you don't need to fetch that since the private key is handed over as script argument (`forge script --rpc-url=... --private-key=... ... --broadcast